modular_zubbers/code/modules/security/security_glock/gun.dm

```diff
@@ -1,28 +1,145 @@
+
+// /obj/item/gun/ballistic/automatic/pistol/sec_glock/security //This is what you give to Security Officers.
+// 	name = "\improper C-CK 9x25mm"
+// 	desc = "The Compact Criminal Killer, or C-CK9 for short, is a semi-automatic ballistic pistol meant for regulated station defense. These are normally issued with a special firing pin that only allows firing on code blue or higher."
+// 	icon = 'modular_zubbers/icons/obj/guns/sec_pistol.dmi'
+// 	icon_state = "black"
+// 	pin = /obj/item/firing_pin/alert_level/blue
+// 	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/flathead
+// 	fire_delay = 4
+// 	can_suppress = FALSE
+// 	projectile_damage_multiplier = 0.65
+
+// /obj/item/gun/ballistic/automatic/pistol/sec_glock/security/rubber //This is what you give to cargo packages.
+// 	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/rubber
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/attack_self(mob/living/user)
+	var/obj/item/storage/our_belt = user.get_item_by_slot(ITEM_SLOT_BELT)
+	var/obj/item/ammo_box/magazine/m9mm/security/swapped_mag = locate(/obj/item/ammo_box/magazine/m9mm/security) in our_belt.contents
+	if(!magazine && swapped_mag)
+		if(swapped_mag)
+			if(insert_magazine(user, swapped_mag))
+				SpinAnimation(2, 1)
+				balloon_alert(user, "you spin your gun, loading in a mag!")
+			else
+				balloon_alert(user, "that magazine doesn't fit!")
+		else
+			balloon_alert(user, "no spare magazines in your belt!")
+		return
+	if(!chambered)
+		if(swapped_mag)
+			SpinAnimation(2, 1) //What a badass
+			if(do_after(user, 1 SECONDS, timed_action_flags = ( IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE ), target = src))
+				throw_eject_magazine(user, magejecting)
+		else
+			balloon_alert(user, "no spare magazines in your belt!")
+	if(bolt_type == BOLT_TYPE_NO_BOLT)
+		unload_ammo(user)
+		return
+	drop_bolt_silent(user)
+	if(recent_rack > world.time)
+		return
+	recent_rack = world.time + rack_delay
+	rack_silent(user)
+	return
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/proc/throw_eject_magazine(mob/user, display_message = TRUE, obj/item/ammo_box/magazine/tac_load = null, atom/target)
+	if(bolt_type == BOLT_TYPE_OPEN)
+		chambered = null
+	if(magazine.ammo_count())
+		playsound(src, eject_sound, eject_sound_volume, eject_sound_vary)
+	else
+		playsound(src, eject_empty_sound, eject_sound_volume, eject_sound_vary)
+	magazine.forceMove(drop_location())
+	var/obj/item/ammo_box/magazine/old_mag = magazine
+	var/obj/item/storage/our_belt = user.get_item_by_slot(ITEM_SLOT_BELT)
+	var/obj/item/ammo_box/magazine/m9mm/security/swapped_mag = locate(/obj/item/ammo_box/magazine/m9mm/security) in our_belt.contents
+	if(swapped_mag)
+		insert_magazine(user, swapped_mag)
+	else
+		magazine = null
+		return
+	old_mag.forceMove(drop_location())
+	if(istype(old_mag, /obj/item/ammo_box/magazine/m9mm/security/rocket) && magejecting)
```

Comment:
Use a var on the magazines to decide what sound you use, I'd say instead of istyping